src/net/sourceforge/kolmafia/session/ResultProcessor.java

```diff
@@ -2071,7 +2071,13 @@ private static void gainItem(boolean adventureResults, AdventureResult result) {
         break;
 
       case ItemPool.GOBLIN_WATER:
+        MonsterData last_monster = MonsterStatusTracker.getLastMonster();
+        if (last_monster == null) {
+          break;
+        }
+        String goblinWaterMonster = last_monster.getName();
         if (adventureResults
+            && goblinWaterMonster.equals("Aquagoblin")
```

Comment:
Shouldn't have to check for Raincore, since AquaGoblin is NOCOPY; if you see one, you have to be in that path.